### DISCUSSION TITLE
Welcome to FabricatedChatReports Discussions!

### DISCUSSION THREAD

**Discussion by FxMorin** [FxMorin/FabricatedChatReports#1] [Announcements]
## 👋 Welcome!
Before you start discussing stuff...

This project was made public due to the lack of information from Mojang and even their disregard for the exploits in 1.19.1-pre2
It's better to release this project publically now in order to prevent these exploits from being possible when 1.19.1 gets released.

Disclaimer: I told the devs beforehand that I would release it if nothing was done about it

**Comment by notabirb1** [comment]
Amazing work! I both hope they patch the exploit / hope they don't at the same time. On one hand, it's a really exploitable thing, but on the other, if left in the game, it could force Mojang to reverse the Chat Reporting.

**Comment by alaajel** [comment]
just wondering, aren't signatures also being sent to mojang to verify? or just straight text only? to check if the player is actually the sender.

**Reply by FxMorin** [reply]
Signatures are also being set to mojang

**Reply by alaajel** [reply]
To be clear, from my current understanding, the client only sends selected messages signatures, but context nope? I unless i am missing something.

**Reply by FxMorin** [reply]
The client sends selected messages and context (7 above, 2 below) for every selected message.
All messages & context include there signatures when sent to mojang

**Reply by alaajel** [reply]
Edit: We've now released 1.19.1 pre-release 4. This pre-release fixes an exploit found regarding contextual chat evidence when creating a Player Report.

pre release 4 released:
```
The order of chat messages are now cryptographically verified
This will be used for validating the context of chat messages for Player Reports
```

**Reply by alaajel** [reply]
so would say exploit got patched

**Reply by FxMorin** [reply]
lol not even close, next pre-release probably
